sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -976,10 +976,26 @@ def test_non_uniform_weights_toy_edge_case_reg():
     y = [0, 0, 1, 0]
     # ignore the first 2 training samples by setting their weight to 0
     sample_weight = [0, 0, 1, 1]
-    for loss in ('ls', 'huber', 'lad', 'quantile'):
-        gb = GradientBoostingRegressor(n_estimators=5)
+    for loss in ('ls', 'lad', 'quantile'):  # not for huber
```

Comment:
added it -- it requires more than one tree; seems like the line search for Huber (called M-regression in the original paper) discounts the original leaf estimate of 1.0 (which is the target value of the only sample in that leaf) to 0.333 . Since the line-search of the Huber loss is an approximation proposed by Friedman, i wonder if that is a result of that. I doubled checked the formula and it coincides with what's written in the paper.
I first suspected an error in the weighted percentile but that looks good.

I now added huber and set the nr of estimators to 2.